### DISCUSSION TITLE
How to use loading.tsx only for page in same route

### DISCUSSION THREAD

**Discussion by JasperAlexander** [vercel/next.js#48954] [Help]
### Summary

How to make sure that only the loading file of the current path is being loaded? I have the following file structure.
- [userid]
    - page.tsx
    - loading.tsx
    - [postid]
        - page.tsx
        - loading.tsx

When a user goes to [userid]/[postid]/page.tsx, [userid]/loading.tsx is being loaded. How to stop this, so that the route only uses [userid]/[postid]/loading.tsx?

**Comment by SamKomesarook** [comment]
Can you post a link to the repo? That setup looks correct..

**Reply by JasperAlexander** [reply]
[https://github.com/JasperAlexander/loading-problem-example](https://github.com/JasperAlexander/loading-problem-example)

If you use throttle Slow 3G in Chrome DevTools and click on the link to grandchild from parent, you're seeing the loading page from child, not grandchild.
